tests/elf/test_static.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+import unittest
+import logging
+import os
+import sys
+import subprocess
+import tempfile
+import shutil
+import lief
+from lief import Logger
+Logger.set_level(lief.LOGGING_LEVEL.INFO)
+
+from subprocess import Popen
+from unittest import TestCase
+
+BINADD_C = """\
+#include <stdio.h>
+#include <stdlib.h>
+
+int add(int a, int b);
+
+int main(int argc, char **argv) {
+  if (argc != 3) {
+    printf("Usage: %s <a> <b>\\n", argv[0]);
+    exit(-1);
+  }
+
+  int res = add(atoi(argv[1]), atoi(argv[2]));
+  printf("From myLIb, a + b = %d\\n", res);
+  return 0;
+}
+"""
+class LibAddSample(object):
+    COUNT = 0
+    def __init__(self):
+        self.logger = logging.getLogger(__name__)
+        self.tmp_dir = tempfile.mkdtemp(suffix='_lief_sample_{:d}'.format(LibAddSample.COUNT))
+        self.logger.debug("temp dir: {}".format(self.tmp_dir))
+
+        LibAddSample.COUNT += 1
+
+        self.binadd_path = os.path.join(self.tmp_dir, "binadd.c")
+        self.binadd_bin = os.path.join(self.tmp_dir, "binadd.o")
+
+        self.compiler = '/usr/bin/cc'
+
+        if self.compiler is None:
+            self.logger.error("Unable to find a compiler")
+            sys.exit(0)
+
+        self.logger.debug("Compiler: {}".format(self.compiler))
+
+        with open(self.binadd_path, 'w') as f:
+            f.write(BINADD_C)
+
+        self._compile_binadd()
+
+
+    def _compile_binadd(self):
+        if os.path.isfile(self.binadd_bin):
+            os.remove(self.binadd_bin)
+
+        cmd = [self.compiler, '-c', '-o', self.binadd_bin, self.binadd_path]
+        self.logger.debug("Compile 'binadd' with: {}".format(" ".join(cmd)))
+        p = Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        stdout, _ = p.communicate()
+        self.logger.debug(stdout)
+
+    @property
+    def binadd(self):
+        return self.binadd_bin
+
+    @property
+    def directory(self):
+        return self.tmp_dir
+
+    def remove(self):
+        if os.path.isdir(self.directory):
+            shutil.rmtree(self.directory)
+
+    def __del__(self):
+        self.remove()
+
+
+class TestStatic(TestCase):
+    def setUp(self):
+        self.logger = logging.getLogger(__name__)
+
+
+    @unittest.skipUnless(sys.platform.startswith("linux"), "requires Linux")
+    def test_write_object(self):
+        sample = LibAddSample()
+        tmp_file = os.path.join(sample.directory, "newfile.o")
+        binadd = lief.parse(sample.binadd)
+        init_obj = [str(o).strip() for o in binadd.object_relocations]
+
+        binadd.write(tmp_file)
+        binadd = lief.parse(tmp_file)
+        new_obj = [str(o).strip() for o in binadd.object_relocations]
+
+        self.assertEqual(len(init_obj), len(new_obj))
+
+        for new, old in zip(new_obj, init_obj):
+            self.assertEqual(new, old)
+
```

Comment:
It would be also nice to check that ``newfile.o`` is still linkable. (e.g. produce the ``libadd.so``)

Reply:
Done